lib/src/span_scanner.dart

```diff
@@ -91,6 +91,17 @@ class SpanScanner extends StringScanner implements LineScanner {
     return _sourceFile.span(startState.position, endPosition);
   }
 
+  /// Creates a [FileSpan] representing the source range between [position] and
+  /// the current position.
```

Comment:
Should the reference to `[position]` be to `[startPosition]` instead? `[position]` and "the current position" mean the same thing to me.

Similarly below, I think "Each [position]" should not use the reference braces.

I think this could also be expanded to give more detail about the different behavior when `endPosition` is null or non-null. I think maybe the first sentence should read something like "between [startPosition] and [endPosition], or the current position if `endPosition` is `null`"?